### ISSUE TITLE
Safari: iOS + macOS: User input is blinking

### ISSUE THREAD

**Issue by LaurensBurger** [open-formulieren/open-forms#3511]
### Product versie / Product version

2.2.1, 2.3.0

### Omschrijf het probleem / Describe the bug
Safari, on both iOS and macOS 
Seems to only occur in forms with BE logic, where fields that are involved in these rules change state the "ping pong" starts. Once these fields are set (by continuing the form) the form "settles".

DH: 592
Browserstack:
![werid](https://github.com/open-formulieren/open-forms/assets/112621537/57408df9-c5b4-41cf-8f14-a5b4579c6e81)

But also testen on physical phone. same behaviour. 
Not limited to DH env

from DH:
![werid2](https://github.com/open-formulieren/open-forms/assets/112621537/500ffb97-ac43-4208-aa16-85f7f9a4f0d3)

something triggers infinite _check_logic requests:
![image](https://github.com/open-formulieren/open-forms/assets/112621537/7ac64d92-6f05-446f-9e70-725712ff114b)

The first time the user data is included, the following request does not, 3rd does, 4th doesn't etc.
1st: 
```json
{
    "data": {
        "organisatie": "hans",
        ...
}
```
2nd
```json
{
    "data": {
        "organisatie": "",
        ...
}
```
3rd: 
```json
{
    "data": {
        "organisatie": "hans",
        ...
}
```

**Comment by sergei-maertens** [comment]
This can be reliably reproduced on any webkit browser - I'm running it native on Epiphany now.

Minimal form to reproduce:

1. Step with two fields (textfield is fine), both not required.
2. Add a logic rule saying that if the value of the first field is not empty, then the second field becomes required.
3. Adding any input to first field triggers the flickering.

The logic causes the formio form definition to change, which triggers form.io re-initializing in the frontend. This is apparently so slow that it keeps toggling back and forth between states.

**Comment by sergei-maertens** [comment]
Tried to reproduce this on 2.1.6 but there the problem doesn't seem to happen. I'm suspecting that the React 18 upgrade in the SDK caused this, as it changed the batching of state updates. This update was done in the 2.2.x release series, which would cause 2.2.x and newer to be affected.

**Comment by sergei-maertens** [comment]
Patch is released as Open Forms 2.2.4 and 2.3.2
